sql/core/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveSessionCatalog.scala

```diff
@@ -423,14 +428,6 @@ class ResolveSessionCatalog(
               s"'${db.head}' != '${v1TableName.database.get}'")
         case _ => ns.map(_.head)
       }
-      if (ns.isDefined && ns.get.length > 1) {
-        throw new AnalysisException(
-          s"Namespace name should have only one part if specified: ${ns.get.quoted}")
-      }
-      if (tbl.length > 2) {
```

Comment:
This is dead code, `al v1TableName = parseV1Table(tbl, sql).asTableIdentifier` fails earlier if `length > 2`